app/assets/src/api/index.js

```diff
@@ -251,7 +251,56 @@ const createProject = params =>
 const logAnalyticsEvent = (eventName, eventData = {}) => {
   // Wrapper around Segment analytics so we can add things later
   // eventData should have keys in snake_case for the database
-  if (window.analytics) window.analytics.track(eventName, eventData);
+  if (window.analytics) {
+    // Include high value user groups in event properties to avoid JOINs downstream.
+    if (window.analytics.user) {
+      const traits = window.analytics.user().traits();
+      eventData = {
+        // see traits_for_segment
+        admin: traits.admin,
+        biohub_user: traits.biohub_user,
+        czi_user: traits.biohub_user,
+        demo_user: traits.demo_user,
+        has_samples: traits.has_samples,
+        // caller can override above traits if they know what they are doing
+        ...eventData
+      };
+    }
+    window.analytics.track(eventName, eventData);
+  }
+};
+
+/**
+ * For wrapping event handlers in React. The first part of the name should be
+ * the React component. The middle part should be user-friendly UI name. The
+ * last part should be the action taken.
+ *
+ * <ReactComponent>_<friendly_name_of_ui_element>_<past_tense_action>
```

Comment:
Actually, I would use a different separator inside and outside of the components if possible: 
`<ReactComponent>_<friendly-name-of-ui-element>_<past_tense_action>` and
`AccessionViz_more-reads-link_clicked`

Reply:
Or alternatively just use camelCase for each component: `AccessionViz_moreReadsLink_clicked`...

Reply:
Yes, using the dash (-) as a separator in the middle part would make it more parseable downstream, but not much, because snowflake and other systems I can imagine us using have a slice function. See https://docs.snowflake.net/manuals/sql-reference/functions/array_slice.html . 

CamelCase on the other hand is not easily parseable and downstream systems may be case insensitive. 

So I'm ambivalent between format such as `AccessionViz_more-reads-link_clicked` and `AccessionViz_more_reads_link_clicked`. I'm happy to go with the most popular.

Reply:
Using the latter would be a problem if you ever need a more complex action name. So, I believe the former is more general and less likely to cause problems in the future.

Reply:
Ok I talked to Mark as well and he preferred the format `AccessionViz_more-reads-link_clicked` so I'll go with that.